Formula/azcopy.rb

```diff
@@ -0,0 +1,16 @@
+class Azcopy < Formula
+  desc "Azure Storage data transfer utility"
+  homepage "https://github.com/Azure/azure-storage-azcopy"
+  url "https://github.com/Azure/azure-storage-azcopy/archive/10.3.4.tar.gz"
+  sha256 "3a0a1450889af252ef251ae5bc86a145f3f77922d316947d7a90088fd1427619"
+
+  depends_on "go" => :build
+
+  def install
+    system "go", "build", *std_go_args
+  end
+
+  test do
+    assert_match "azcopy version", shell_output("#{bin}/azcopy --version")
```

Comment:
We need a test that exercises the some of the functionality of the app. Version checks or usage checks (foo --version or foo --help) are not sufficient, as explained in the formula cookbook.

In most cases, a good test would involve running a simple test case: run #{bin}/foo input.txt.

- Then you can check that the output is as expected (with assert_equal or assert_match on the output of shell_output)
- You can also check that an output file was created, if that is expected: assert_predicate testpath/"output.txt", :exist?

Some advice for specific cases:

- If the formula is a library, compile and run some simple code that links against it. It could be taken from upstream's documentation / source examples.
- If the formula is for a GUI program, try to find some function that runs as command-line only, like a format conversion, reading or displaying a config file, etc.
- If the software cannot function without credentials, a test could be to try to connect with invalid credentials (or without credentials) and confirm that it fails as expected.
- Same if the software requires a virtual machine, docker instance, etc. to be running.

Reply:
Hi @SMillerDev,

I understood your point, however, for doing some test exercises it will require a public storage account on the Azure side.

Is it possible to leave as is till a volunteer provides us a public storage account on Azure? 

Thanks.
Eduardo.

Reply:
No, not having proper tests is unacceptable. The suggestions provide a test case for such a scenario though.

Reply:
ok. now looks better. just added a test.